WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationSettingDetailsViewController.swift

```diff
@@ -0,0 +1,143 @@
+        // Register the cells
+        tableView.registerClass(SwitchTableViewCell.self, forCellReuseIdentifier: reuseIdentifier)
+                
+        // Hide the separators, whenever the table is empty
+        tableView.tableFooterView = UIView()
+        
+        // Style!
+        WPStyleGuide.configureColorsForView(view, andTableView: tableView)
+    }
+    
+    
+    
+    // MARK: - Public Helpers
+    public func setupWithSettings(settings: NotificationSettings, stream: NotificationSettings.Stream) {
+        // Title
+        switch settings.channel {
+        case .WordPressCom:
+            title = NSLocalizedString("WordPress.com Updates", comment: "WordPress.com Notification Settings Title")
+        default:
+            title = stream.kind.description()
+        }
+        
+        // Structures
+        self.settings   = settings
+        self.stream     = stream
+        self.newValues  = [String: Bool]()
+        
+        tableView.reloadData()
+    }
+    
+    
+    
+    // MARK: - UITableView Delegate Methods
+    public override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return sectionCount
+    }
+    
+    public override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return settings?.sortedPreferenceKeys(stream).count ?? emptyRowCount
+    }
+    
+    public override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithIdentifier(reuseIdentifier) as! SwitchTableViewCell
+        
+        configureCell(cell, indexPath: indexPath)
+        
+        return cell
+    }
+    
+    
+    
+    // MARK: - UITableView Helpers
+    private func configureCell(cell: SwitchTableViewCell, indexPath: NSIndexPath) {
+        let preferences         = stream?.preferences
+        let sortedKeys          = settings?.sortedPreferenceKeys(stream)
+        let key                 = sortedKeys?[indexPath.row]
+        if preferences == nil || key == nil {
+            return
+        }
+        
+        cell.name               = settings?.localizedDescription(key!) ?? String()
+        cell.isOn               = preferences?[key!] ?? true
+        cell.onChange           = { [weak self] (newValue: Bool) in
+            self?.newValues?[key!] = newValue
+        }
+    }
+    
+    
+    
+    // MARK: - Service Helpers
+    private func saveSettingsIfNeeded() {
+        if newValues?.count == 0 || settings == nil {
+            return
+        }
+
+        let context = ContextManager.sharedInstance().mainContext
+        let service = NotificationsService(managedObjectContext: context)
+                
+        service.updateSettings(settings!,
+            stream              : stream!,
+            newValues           : newValues!,
+            success             : nil,
+            failure             : { (error: NSError!) in
```

Comment:
In this particular case, the strong `self` reference is on purpose: the error handling itself will retry if something went wrong. The entire instance will be released either when the user `Cancels` the retry, or the update actually succeeds.